### PR TITLE
Feature/BSA-272/Allow to hide "Logout" and "Home" buttons for the delivery execution

### PR DESCRIPTION
**Pull request by shpran**
Relates to: [BSA-272](https://oat-sa.atlassian.net/browse/BSA-272)

**Changes:**
* added possibility to configure the visibility of the "Logout" and "Home" buttons for the delivery execution header;
* added default configuration to show this buttons.

**Companion PR:** [#73](https://github.com/oat-sa/extension-tao-bosa-selor/pull/73)

### CODE REVIEW

**Review by bziondik — CHANGES_REQUESTED** [review]
Have error on Update
image

**Review by bziondik — APPROVED** [review]
Checked locally, works as expected

File: scripts/tools/ConfigureDeliveryExecutionHeader.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+use common_report_Report as Report;
+use common_exception_Error as Error;
+use oat\oatbox\extension\script\ScriptAction;
+use common_ext_ExtensionsManager as ExtensionsManager;
+use common_ext_ExtensionException as ExtensionException;
+use oat\oatbox\service\exception\InvalidServiceManagerException;
+
+/**
+ * Class ConfigureDeliveryExecutionHeader
+ *
```
Comment (SergiiTao) [inline scripts/tools/ConfigureDeliveryExecutionHeader.php:34]: very meaningful docblock :)

File: scripts/tools/ConfigureDeliveryExecutionHeader.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+     * @return Report
+     */
+    protected function run()
+    {
+        /** @var ExtensionsManager $extensionManager */
+        $extensionManager = $this->getServiceManager()->get(ExtensionsManager::SERVICE_ID);
+        $extension = $extensionManager->getExtensionById('taoDelivery');
+
+        $config = $extension->getConfig(self::CONFIG_KEY);
+        $config['hideHomeButton'] = $this->getOption(self::OPTION_HIDE_HOME_BUTTON);
```
Comment (SergiiTao) [inline scripts/tools/ConfigureDeliveryExecutionHeader.php:89]: I think we can use constants as keys also, right?

Comment (shpran) [reply scripts/tools/ConfigureDeliveryExecutionHeader.php:89]: Right :)
Thanks for the remark

**Review by SergiiTao — CHANGES_REQUESTED** [review]
minor cs fixes

File: migrations/Version202010131343168613_taoDelivery.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+use Doctrine\DBAL\Schema\Schema;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use oat\taoDelivery\scripts\tools\ConfigureDeliveryExecutionHeader;
+
+/**
+ * Class Version202010131343168613_taoDelivery
+ *
+ * @package oat\taoDelivery\migrations
+ */
+final class Version202010131343168613_taoDelivery extends AbstractMigration
```
Comment (shpran) [inline migrations/Version202010131343168613_taoDelivery.php:34]:
```suggestion
final class Version202010131343168613_taoDelivery extends AbstractMigration
```

File: scripts/tools/ConfigureDeliveryExecutionHeader.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+use common_ext_ExtensionException as ExtensionException;
+use oat\oatbox\service\exception\InvalidServiceManagerException;
+
+/**
+ * Class ConfigureDeliveryExecutionHeader
+ * Script allow to control the visibility of the "Home" and "Logout" buttons while executing the delivery.
+ *
+ * @package oat\taoDelivery\scripts\tools
+ */
+class ConfigureDeliveryExecutionHeader extends ScriptAction
```
Comment (shpran) [inline scripts/tools/ConfigureDeliveryExecutionHeader.php:38]:
```suggestion
class ConfigureDeliveryExecutionHeader extends ScriptAction
```

File: scripts/tools/ConfigureDeliveryExecutionHeader.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+namespace oat\taoDelivery\scripts\tools;
+
+use common_report_Report as Report;
+use common_exception_Error as Error;
+use oat\oatbox\extension\script\ScriptAction;
+use common_ext_ExtensionsManager as ExtensionsManager;
+use common_ext_ExtensionException as ExtensionException;
+use oat\oatbox\service\exception\InvalidServiceManagerException;
+
+/**
```
Comment (shpran) [inline scripts/tools/ConfigureDeliveryExecutionHeader.php:32]:
```suggestion
```

File: controller/DeliveryServer.php [hunk 6]
```diff
@@ -253,13 +254,19 @@ public function runDeliveryExecution()
 
         $this->setData('additional-header', $container->getContainerHeader());
         $this->setData('container-body', $container->getContainerBody());
-        
+
+        /** @var ExtensionsManager $extensionManager */
+        $extensionManager = $this->getServiceLocator()->get(ExtensionsManager::SERVICE_ID);
+        $extension = $extensionManager->getExtensionById('taoDelivery');
+        $headerConfig = $extension->getConfig('deliveryExecutionHeader');
```
Comment (SergiiTao) [inline controller/DeliveryServer.php:261]: please consider using a proper config repository with validation. 
looking at `$headerConfig = $extension->getConfig('deliveryExecutionHeader');` it is not obvious what is the type, what is inside, is it valid, etc.

**Review by SergiiTao — CHANGES_REQUESTED** [review]

File: controller/DeliveryServer.php (hunk 6 above)
Comment (shpran) [reply controller/DeliveryServer.php:261]: Do you have any suggestions?

Comment (SergiiTao) [reply controller/DeliveryServer.php:261]: `class DeliveryExecutionConfig extends ConfigurableService
public function isHomeButtonHidden(): bool;
...
`
or may be we already have something like this

Comment (shpran) [reply controller/DeliveryServer.php]: Done

File: model/execution/DeliveryExecutionConfig.php [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+    public const OPTION_HIDE_HOME_BUTTON = 'hideHomeButton';
+    public const OPTION_HIDE_LOGOUT_BUTTON = 'hideLogoutButton';
+
+    /**
+     * @return bool
+     */
+    public function isHomeButtonHidden(): bool
+    {
+        return (bool) $this->getOption(self::OPTION_HIDE_HOME_BUTTON, false);
```
Comment (SergiiTao) [inline model/execution/DeliveryExecutionConfig.php:39]: as in the previous PR 
(bool) doesn't guarantees a correct value. please consider validating input values and cast types in a safer way

**Review by SergiiTao — CHANGES_REQUESTED** [review]

File: model/execution/DeliveryExecutionConfig.php [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        $this->setOption(self::OPTION_HIDE_LOGOUT_BUTTON, $hide);
+    }
+
+    /**
+     * @param $value
+     *
+     * @return bool
+     */
+    public static function toBool($value): bool
```
Comment (SergiiTao) [inline model/execution/DeliveryExecutionConfig.php:71]: don't reinvent a wheel 
`filter_var($var, FILTER_VALIDATE_BOOLEAN)` or `filter_var($var, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE)` works just fine

**Review by SergiiTao — CHANGES_REQUESTED** [review]

File: model/execution/DeliveryExecutionConfig.php (hunk 8 above)
Comment (shpran) [reply model/execution/DeliveryExecutionConfig.php:71]: Oh, forgot about this function, thanks

**Review by SergiiTao — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**